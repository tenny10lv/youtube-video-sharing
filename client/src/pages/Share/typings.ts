export enum Field {
  LINK = "link",
}

export type FormField = {
  [Field.LINK]: string;
};
