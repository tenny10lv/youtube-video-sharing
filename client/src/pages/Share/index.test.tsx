import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Share from "./";
import { ToastContainer } from "react-toastify";

jest.mock("../../context/AuthContext", () => {
  return {
    useAuth: () => ({
      user: {
        id: 1,
      },
    }),
  };
});

jest.mock("../../api/movies");
const { shareMovie, getMovieDetail } = require("../../api/movies");

test("Should match snapshot", () => {
  const { container } = render(<Share />);
  expect(container).toMatchSnapshot();
});

test("renders share title", () => {
  render(<Share />);
  const title = screen.getByText(/Share A Movie/i);
  expect(title).toBeInTheDocument();
});

test("share url successfully", async () => {
  shareMovie.mockImplementation(() => ({
    data: {},
  }));
  getMovieDetail.mockImplementation(() => ({}));

  render(
    <>
      <ToastContainer autoClose={3000} position="bottom-right" />
      <Share />
    </>
  );
  const url = "https://www.youtube.com/watch?v=chdLDqOfcwk";

  const input = screen.getByLabelText("Youtube URL") as any;
  fireEvent.change(input, {
    target: { value: url },
  });
  expect(input.value).toBe(url);

  const shareButton = screen.getByRole("button", {
    name: "Share",
  });
  expect(shareButton).toBeInTheDocument();

  fireEvent.click(shareButton);

  expect(await screen.findByText("Shared movie successfully")).toBeVisible();
});

test("share url invalid link", async () => {
  render(
    <>
      <ToastContainer autoClose={3000} position="bottom-right" />
      <Share />
    </>
  );
  const url = "https://www.youtube.com/watch?v=9SnAO_QYRG";

  const input = screen.getByLabelText("Youtube URL") as any;
  fireEvent.change(input, {
    target: { value: url },
  });
  expect(input.value).toBe(url);

  const shareButton = screen.getByRole("button", {
    name: "Share",
  });
  expect(shareButton).toBeInTheDocument();

  fireEvent.click(shareButton);

  expect(await screen.findByText("The link is invalid")).toBeVisible();
});

test("share url not found movie", async () => {
  getMovieDetail.mockImplementation(() => undefined);

  render(
    <>
      <ToastContainer autoClose={3000} position="bottom-right" />
      <Share />
    </>
  );
  const url = "https://www.youtube.com/watch?v=9SnAO_QYRGi";

  const input = screen.getByLabelText("Youtube URL") as any;
  fireEvent.change(input, {
    target: { value: url },
  });
  expect(input.value).toBe(url);

  const shareButton = screen.getByRole("button", {
    name: "Share",
  });
  expect(shareButton).toBeInTheDocument();

  fireEvent.click(shareButton);

  expect(await screen.findByText("Not found movie")).toBeVisible();
});
