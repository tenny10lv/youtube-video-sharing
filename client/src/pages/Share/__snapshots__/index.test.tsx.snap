// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-vgft9y"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-sthl0-MuiTypography-root"
      >
        Share A Movie
      </h4>
      <form
        class="MuiBox-root css-8atqhb"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Youtube URL
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="link"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Youtube URL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Share
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          href="/"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-61ksmc-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Back to list
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </form>
    </div>
  </div>
</div>
`;
